### ISSUE TITLE
Error when using React.createElement directly

### ISSUE THREAD

**Issue by mjomble** [facebook/flow#4658]
I'm not a big fan of JSX and prefer to use React.createElement directly.

[The Babel REPL link](http://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact&experimental=false&loose=false&spec=false&code=const%20element%20%3D%20%3Ch1%3EHello%2C%20world!%3C%2Fh1%3E%3B%0Aconst%20container%20%3D%20document.getElementById('root')%3B%0AReactDOM.render(element%2C%20container)%3B%0A) from the ["Hello World" page from React docs](https://facebook.github.io/react/docs/hello-world.html) shows `<h1>Hello, world!</h1>` transpiling to

```
React.createElement(
  'h1',
  null,
  'Hello, world!'
)
```

If I [try this in Try Flow](https://flow.org/try/#0JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wChl0YA6NIlGJAUQBskQkA7GACnwAsAjPgA0cDgFdmzUfgASSKRFEB3aMwAmAQnwBKIA) with 0.53.1, I get the following errors:

```
2: React.createElement('h1', null, 'Hello, world!')
                             ^ React key. Property not found in possibly null value
2: React.createElement('h1', null, 'Hello, world!')
                             ^ null
2: React.createElement('h1', null, 'Hello, world!')
                             ^ React ref. Property not found in possibly null value
2: React.createElement('h1', null, 'Hello, world!')
                             ^ null
```

However, `React.createElement('h1', {}, 'Hello, world!')` works fine.

Is this a bug in Flow or expected behavior?

**Comment by apalm** [comment]
It looks like this is still an issue in 0.54.0: 
https://flow.org/try/#0PTAEAEDMBsHsHcBQiCWBbADrATgF1AFSgCGAzqAEoCmxAxvpNrGqAETY32sDcyuAnhiqgAYrFgAFJhnIBeUAG8APogCQtABYpoAEw4A7AFyhSubCn0BzREoC+yWrH2nR449Tq4AdAGVcxXCpoKlJSEQBXfXoUJ2JoAGFmLH0qfVwAHjFJaVIAPlB5DByC3LV0nRQAN1yi2BkvTW09VPTgCuqHJxcAIWJsd05vPwCgkLDI6NiEpKdUjIVbfPkACgBKErKs3IAjPtatzud8XoAvAc9ff0Dg0Iio3Bj9OMTMWbT0haXQNY3VD3oGhwRgBRYJoObLLIAGlA+nC0GgMIA5LsTkjVkA

**Comment by asolove** [comment]
Looks like this was closed incorrectly. A PR mentioned this issue number, but not because it fixed it. Instead because it was adding lots of test ignore comments caused by this issue. Seems like this should get fixed! (/cc @calebmer)

**Comment by SimenB** [comment]
@asolove it got closed by https://github.com/facebook/flow/commit/30f8e7b1c548702587d8e9c7972a7194a0cd0fca, which has "fixes #4658 in the title"

**Comment by nmote** [comment]
The fix didn't make the cut for 0.54; it will come out with 0.55.0. Feel free to reopen if the issue persists after that.
